.github/workflows/reward.yml

```diff
@@ -0,0 +1,23 @@
+name: Add Github User to Allowlist on PR Merge
+
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  allowlist_user:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - name: Use Allowlist GitHub Action
+        uses: ton-society/github-allowlist-action@main
```

Comment:
using `@main` is unsafe and can lead to all your secrets leak if `github-allowlist-action@main` is compromised. please, use a tag